cEP-0035.md

```diff
@@ -0,0 +1,187 @@
+| Status   | Proposed                              |
+| Type     | Process                               |
+
+## Abstract
+
+This cEP describes the upgrades and steps required to enable coala to support newer versions of Python. This project has been taken up as a [GSoC'21](https://summerofcode.withgoogle.com/organizations/6490386047959040/#4634486723051520) project.
+
+## Introduction
+
+coala, as of now, supports python up to version 3.6, which is quite old. Lack of support of newer versions of python has been a significant hurdle in the development of coala. Version 3.6 was released back in 2016, and python has made a considerable amount of progress since then. coala, on the other hand, has largely not been able to benefit from these upgrades because of the strict version limitations. Upgrading coala to support newer versions will accelerate development of coala and will make it future proof.
+
+## Objectives
+
+TThe project's primary goal is to enable coala to support newer versions of Python, but this will be attained in multiple smaller steps. The goal is not just to facilitate support for newer versions of Python but also to upgrade the coala to support more recent dependencies. Upgrades will be performed over multiple repositories as the dependencies across the project are closely interlinked. Will perform changes in coala, bears, dependency_mangement, docker, mobans and coala-utils repository. 
+
+## Proposed Approach
+
+### Keeping the CI/CD pipeline green
+
+Throughout the project, it needs to be ensured that the CI/CD pipeline remains unbroken as it is crucial for running the test suite, which is necessary for PR to get merged. This can be ensured by making commits keeping into consideration the pipeline. And make changes over multiple repositories before merging, in case required to keep the pipeline running.
+Currently, the bears CI is broken, so I will start by fixing it.
+
+### Upgrade dependency_management system 
+
+The dependency_management system provides a Python API for familiar package managers. It provides a generic method to install dependencies in the host operating system. It has been a while since it has been upgraded. Thus it supports only the earlier versions of package managers. Will upgrade the Python, dependencies and versions of the package managers to access newer dependencies using the API. We will enable support for Python3.9 and then move forward, upgrading the dependencies to their latest versions.
+
+### Upgrade bears
+
+Currently bears don't support newer pip versions, changes will be made to fix that. This is essential as while installing the latest version of a Python package might fail or install at a slower rate. This will be followed by upgrade in existing dependencies and Python version. Some changes might be required in mobans and coala as well to keep the pipeline up running. The problems associated with bears and their probable solutions are:
```

Comment:
Please improve the wordings as mentioned in previous points, not mentioning them again to reduce clutter. Just try to give factual info here.